csunplugged/tests/topics/loaders/test_lessons_loader.py

```diff
@@ -0,0 +1,457 @@
+        lessons_structure = self.test_data.load_yaml_file(complete_config_file_path)
+
+        topic = self.test_data.create_topic("1")
+        unit_plan = self.test_data.create_unit_plan(topic, "1")
+
+        lesson_loader = LessonsLoader(
+            config_file,
+            self.load_log,
+            lessons_structure,
+            topic,
+            unit_plan,
+            self.test_data.LOADER_ASSET_PATH
+        )
+        lesson_loader.load()
+
+        self.assertQuerysetEqual(
+            Lesson.objects.all(),
+            ["<Lesson: Lesson 1>"]
+        )
+
+    def test_min_age(self):
+        config_file = os.path.join(self.loader_name, "basic-config.yaml")
+        complete_config_file_path = os.path.join(self.test_data.LOADER_ASSET_PATH, config_file)
+        lessons_structure = self.test_data.load_yaml_file(complete_config_file_path)
+
+        topic = self.test_data.create_topic("1")
+        unit_plan = self.test_data.create_unit_plan(topic, "1")
+
+        lesson_loader = LessonsLoader(
+            config_file,
+            self.load_log,
+            lessons_structure,
+            topic,
+            unit_plan,
+            self.test_data.LOADER_ASSET_PATH
+        )
+        lesson_loader.load()
+        self.assertEquals(
+            Lesson.objects.get(slug="lesson-1").min_age,
+            1,
+        )
+
+    def test_max_age(self):
+        config_file = os.path.join(self.loader_name, "basic-config.yaml")
+        complete_config_file_path = os.path.join(self.test_data.LOADER_ASSET_PATH, config_file)
+        lessons_structure = self.test_data.load_yaml_file(complete_config_file_path)
+
+        topic = self.test_data.create_topic("1")
+        unit_plan = self.test_data.create_unit_plan(topic, "1")
+
+        lesson_loader = LessonsLoader(
+            config_file,
+            self.load_log,
+            lessons_structure,
+            topic,
+            unit_plan,
+            self.test_data.LOADER_ASSET_PATH
+        )
+        lesson_loader.load()
+        self.assertEquals(
+            Lesson.objects.get(slug="lesson-1").max_age,
+            99,
+        )
+
+    def test_number(self):
+        config_file = os.path.join(self.loader_name, "basic-config.yaml")
+        complete_config_file_path = os.path.join(self.test_data.LOADER_ASSET_PATH, config_file)
+        lessons_structure = self.test_data.load_yaml_file(complete_config_file_path)
+
+        topic = self.test_data.create_topic("1")
+        unit_plan = self.test_data.create_unit_plan(topic, "1")
+
+        lesson_loader = LessonsLoader(
+            config_file,
+            self.load_log,
+            lessons_structure,
+            topic,
+            unit_plan,
+            self.test_data.LOADER_ASSET_PATH
+        )
+        lesson_loader.load()
+        self.assertEquals(
+            Lesson.objects.get(slug="lesson-1").number,
+            1,
+        )
+
+    def test_topic(self):
```

Comment:
I think this one also needs a better name. I spent a while trying to decipher what this did.